Source/Core/CustomProjection.js

```diff
@@ -0,0 +1,311 @@
+define([
+        './Cartesian3',
+        './Cartographic',
+        './Check',
+        './defaultValue',
+        './defined',
+        './defineProperties',
+        './DeveloperError',
+        './Ellipsoid',
+        './getAbsoluteUri',
+        './isDataUri',
+        './loadAndExecuteScript',
+        '../ThirdParty/when'
+    ], function(
+        Cartesian3,
+        Cartographic,
+        Check,
+        defaultValue,
+        defined,
+        defineProperties,
+        DeveloperError,
+        Ellipsoid,
+        getAbsoluteUri,
+        isDataUri,
+        loadAndExecuteScript,
+        when) {
+    'use strict';
+
+    /**
+     * {@link MapProjection} that uses custom project and unproject functions defined in an external file.
```

Comment:
I think this description could be improved.  The fact that it uses an external file is an implementation requirement, not something that defines this class.  The class itself exists to allow for completely custom projections not supported by Proj4JS or either of the other standard Map projections.  We should also add `@see` tags to those in this description as well as a `@demo` tag to the unified Sandcastle example I suggested.  (We should do similar things in the Proj4 projection as well).

Reply:
We could have the users implement `MapProjection`, but that could be messy since the `MapProjection` base class exposes a Cesium ellipsoid and has a private flag for whether or not Cesium can make the assumptions it makes for geographic and web mercator.

The original intention was to get as close as possible to just requiring that the user provide `project` and `unproject` callbacks and still have something that can be delivered to a web worker.

Reply:
Makes sense.